git_repo/services/service.py

```diff
@@ -244,7 +244,10 @@ def format_path(self, repository, namespace=None, rw=False):
         if not rw and '/' in repo:
             return '{}/{}'.format(self.url_ro, repo)
         elif rw and '/' in repo:
-            return '{}:{}'.format(self.url_rw, repo)
+            if self.url_rw.startswith('ssh://'):
```

Comment:
GitBucket's ssh url is:  `ssh://git@server:29418/root/repo.git`

GitBucket doesn't use system sshd. It uses embedded sshd. So, it can't listen on 22 port.
By [Git manual](https://git-scm.com/docs/git-clone#_git_urls_a_id_urls_a), "An alternative scp-like syntax" doesn't take ssh port. So it is required for GitBucket with ssh repository access.